src/interface/update.hpp

```diff
@@ -150,6 +151,73 @@ TaskStatus FillDerived(T *rc) {
   return TaskStatus::complete;
 }
 
+template <typename T>
+TaskStatus InitNewlyAllocatedVars(T *rc) {
+  if (!rc->AllVariablesInitialized()) {
+    std::cout << "Initializing variable interiors after allocation" << std::endl;
+
+    const IndexDomain interior = IndexDomain::interior;
+    const IndexRange ib = rc->GetBoundsI(interior);
+    const IndexRange jb = rc->GetBoundsJ(interior);
+    const IndexRange kb = rc->GetBoundsK(interior);
```

Comment:
You potentially have received data from other blocks in your boundaries that is not unitialized.

Reply:
Is there a case where this may lead to problems (e.g., because reconstructed, but uninitialized boundary data is being used to update some active zones)?

Reply:
The boundaries that are receiving data are filled and the boundaries that are not receiving data are filled with the default value in `ReceiveBoundBufs`, so everything should be initialized.